chisel/src/cmd_oneliner.rs

```diff
@@ -0,0 +1,120 @@
+                Some(p) => p.to_string(),
+                None => "/dev/stdout".to_string(),
+            };
+
+            let chisel_config = match ChiselConfig::from_args(module_list, options_list) {
+                Ok(mut config) => {
+                    // Inject the input and output file paths here.
+                    config.rulesets_mut()[0]
+                        .1
+                        .options_mut()
+                        .insert("file".to_string(), file.to_string());
+                    config.rulesets_mut()[0]
+                        .1
+                        .options_mut()
+                        .insert("output".to_string(), output);
+                    config
+                }
+                Err(e) => fail(1, &format!("Failed to load configuration: {}", e)),
+            };
+
+            chisel_debug!(1, "{}", chisel_config);
+
+            let mut driver = ChiselDriver::new(chisel_config);
+
+            let mut exit_code = 0;
+            loop {
+                match driver.fire() {
+                    DriverState::Error(err, _) => {
+                        fail(1, &format!("runtime error: {}", err));
+                    }
+                    DriverState::Done(_) => break,
+                    _ => panic!("Should never return READY"),
+                }
+            }
+
+            let mut results = driver.take_result();
+            // wish list: write yaml-encoded results to stdout
+            chisel_debug!(1, "Module execution completed successfully");
+            eprintln!("{}", &results);
+
+            // Get ruleset
+            let results = results.rulesets_mut();
+            let io_result = match flags.value_of("output.mode") {
+                Some("bin") => {
+                    let mut result = results.pop().expect("One ruleset was executed");
+                    result.write("bin")
+                }
+                #[cfg(feature = "wabt")]
+                Some("wat") => {
+                    let mut result = results.pop().expect("One ruleset was executed");
+                    result.write("wat")
+                }
+                #[cfg(not(feature = "wabt"))]
+                Some("wat") => Err("This build does not support wat-encoded output"
+                    .to_string()
+                    .into()),
+                Some("hex") => {
+                    let mut result = results.pop().expect("One ruleset was executed");
+                    result.write("hex")
+                }
+                _ => panic!("CLI parser ensures value can only be one of the above"),
+            };
+
+            match io_result {
+                Ok(true) => eprintln!("Successfully wrote output to file."),
+                Ok(false) => eprintln!("No changes to write."),
+                Err(e) => {
+                    eprintln!("failed to write output to file: {}", e.description());
+                    exit_code = 1;
+                }
+            }
+
+            exit_code
```

Comment:
I think this could be removed just returning the code in the match? But at least moving the `exit_code` declaration in before this match. Where it is before the loop is too early.